zooniverse/Panoptes-Front-End
# Project Builder: remove CoffeeScript from Multi Image Subject Options Editor

## PR Overview

This PR converts the Multi Image Subjects Options Editor from CoffeeScript to JavaScript. _No behaviour changes are expected._

Context: FEM projects will start supporting the Flipbook feature on the FEM Classifier, and as a result, the FEM Lab now needs to support configuring those options. At the moment, all "Multi Image Options" on FEM Lab are deprecated, so in preparation for future development, we need to decaffeinate the component.

### Testing

Staging branch URL: https://pr-{NUMBER}.pfe-preview.zooniverse.org

Testing steps:
- Open the Project Builder, choose a project, create a workflow. Ensure the WF Editor is in PFE Lab mode. [(Example)](https://master.pfe-preview.zooniverse.org/lab/1986/workflows/3714?pfeLab=true&env=staging)
- Default values: when the workflow is created, observe that the "Multi-Image Options" section has default values.
- Changing values: start modifying values on each option.
  - Reload the page to ensure the changed values stick.
  - Note that "iterations" defaults to 3, but deleting the value (setting it to blank) will set an implicit value of infinity
  - Note that setting layout to "Show separate frames" will display additional options for layout.

Additional Classifier checks:
- Try viewing the newly-created workflow on the PFE Classifier. All options should work as expected (Assumes you have multi-image Subjects.)

Additional FEM Lab checks:
- The newly-created workflow should be viewable in FEM Lab mode. [(Example)](https://master.pfe-preview.zooniverse.org/lab/1986/workflows/3714?femLab=true&env=staging)
  - The "Multi-Image Options" but should be deprecated (not editable/non-interactive), but the values should still be visible

### Status

Ready for review.

A follow up will be required to add FEM Lab-specific options to the Multi Image Subject Options Editor.
